Source/Core/Core/PatchEngine.cpp

```diff
@@ -277,9 +281,6 @@ static void ApplyPatches(const Core::CPUThreadGuard& guard, const std::vector<Pa
 static void ApplyMemoryPatches(const Core::CPUThreadGuard& guard,
                                std::span<const std::size_t> memory_patch_indices)
 {
-  if (AchievementManager::GetInstance().IsHardcoreModeActive())
-    return;
-
```

Comment:
You're not filtering memory patches anywhere (and I think they can only come from a debugger...?) so you probably want to keep blocking here.